### ISSUE TITLE
Automatically assign the efi mountpoint to the EFI partition

### ISSUE THREAD

**Issue by ghost** [calamares/calamares#1714]
**Is your feature request related to a problem? Please describe.**
Currently, Calamares doesn't automatically assign the EFI partition with the efi mountpoint.

**Describe the solution you'd like**
Since the EFI partition is a fat32 partition and has the boot flag, Calamares could detect these two indicators and assign the efi mountpoint to it automatically, without formatting it of course (so it doesn't break dual booting).

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by demmm** [comment]
>assign /boot/efi mount point to it automatically
That is not correct to use as an automated mountpoint, not all systems use /boot/efi for ESP (see all the systemd-boot docs regarding this).

**Comment by lyrebird-sound-cloning-system** [comment]
Understood. I edited the title.

**Comment by adriaandegroot** [comment]
This applies to manual partitioning, I suppose -- you should not see or notice the EFI partition when doing erase, alongside or repace installs because it picks them up automatically. I think a scenario describing user actions and expected behavior would help here.

The code is in `src/modules/partition/gui/CreatePartitionDialog.cpp`, and it *might* be enough to detect changes to the flags -- there will be some messing about in `setFlagList()` as well, but it looks doable. I just don't have an intuition about what to do if the user chooses a different point for a partition with the boot flag, or what the mount point should become if the boot flag is toggled off.

**Comment by geckolinux** [comment]
Possibly related request?
https://github.com/calamares/calamares/issues/1111

**Comment by ezsit55** [comment]
Please do not implement any form of auto-detection and assignment of a mount point of the ESP during manual partitioning. Manual should remain truly manual. There is no need to insert more complex calculations into a module that provides a manual method to partition disks. The user who chooses manual partitioning does not need the installer to guess at the possible ESP. Any code that attempts to guess at the ESP and assign a mount point will invariably introduce new bugs and errors, and for no good reason. A manual partitioning mechanism exists for the user who wants total control and not have the system make guesses and decisions that are possibly not what the user wants.

**Comment by adriaandegroot** [comment]
The **logic** would be simple, I suppose:
- if the system is EFI, and
- the *boot* flag is switched on, and
- no mount point is selected yet, and
- the partition is set to fat32, and
- no other partition has the EFI mount point, and
- no other partition has the *boot* flag
- **then** set the mount point to the EFI mount point.

That's not really simple logic, and it introduces a mess of other complications, like: what if the "boot" flag is then toggled off again? The UI flow doesn't suggest this either: top-to-bottom you have partition type, then mount point, then flags. To hit this special case, you'd have to be jumping around the UI **and** have a pressing need to save two clicks in an already complicated setup.

(The issue mentioned by @geckolinux is .. perhaps I misunderstood this to be a request about manual partitioning where, as I've just described, it wouldn't fit; for automatic partitioning, see that issue. Closing this one as "submitter is gone, needs-feedback, probably a dup".)
